// clinical-nutrition/app/components/SearchModal.tsx
"use client";

import { useState, useEffect, useRef } from "react";
import Link from "next/link";
import { useRouter } from "next/navigation";
import { Search, X, BookOpen, Calculator } from "lucide-react";
import { articles, searchArticles, type ArticleMetadata } from "../../lib/content/articles";

interface SearchResult {
  title: string;
  href: string;
  category: "knowledge" | "tool";
  description?: string;
  tags?: string[];
}

// Tools search index (still hardcoded as they're not in articles metadata)
const toolsIndex: SearchResult[] = [
  // Tools
  {
    title: "Sàng lọc Suy Dinh dưỡng",
    href: "/tools/malnutrition-screening",
    category: "tool",
    description: "Công cụ sàng lọc nguy cơ suy dinh dưỡng",
  },
  {
    title: "Máy tính Năng lượng & Protein",
    href: "/tools/energy-protein-calculator",
    category: "tool",
    description: "Tính toán nhu cầu năng lượng và protein",
  },
  {
    title: "Đánh giá Lượng Ăn",
    href: "/tools/intake-assessment",
    category: "tool",
    description: "Đánh giá lượng dinh dưỡng thực tế",
  },
  {
    title: "BMI & Thành phần Cơ thể",
    href: "/tools/bmi-body-composition",
    category: "tool",
    description: "Tính BMI, IBW, ABW, BSA",
  },
  {
    title: "Nhu cầu Dịch",
    href: "/tools/fluid-requirements",
    category: "tool",
    description: "Tính nhu cầu dịch hàng ngày",
  },
];

function highlightText(text: string, query: string): React.ReactNode {
  if (!query.trim()) return text;

  const parts = text.split(new RegExp(`(${query})`, "gi"));
  return (
    <>
      {parts.map((part, i) =>
        part.toLowerCase() === query.toLowerCase() ? (
          <mark key={i} className="bg-yellow-200 font-medium">
            {part}
          </mark>
        ) : (
          part
        )
      )}
    </>
  );
}

interface SearchModalProps {
  onClose: () => void;
}

export function SearchModal({ onClose }: SearchModalProps) {
  const [query, setQuery] = useState("");
  const [results, setResults] = useState<SearchResult[]>([]);
  const inputRef = useRef<HTMLInputElement>(null);
  const router = useRouter();

  useEffect(() => {
    inputRef.current?.focus();
  }, []);

  useEffect(() => {
    if (!query.trim()) {
      setResults([]);
      return;
    }

    const lowerQuery = query.toLowerCase();
    
    // Search articles from metadata system
    const articleResults = searchArticles(query).map((article): SearchResult => ({
      title: article.title,
      href: article.href,
      category: "knowledge" as const,
      description: article.description,
      tags: article.tags,
    }));

    // Search tools
    const toolResults = toolsIndex.filter(
      (item) =>
        item.title.toLowerCase().includes(lowerQuery) ||
        item.description?.toLowerCase().includes(lowerQuery)
    );

    // Combine and sort by relevance (title matches first)
    const allResults = [...articleResults, ...toolResults].sort((a, b) => {
      const aTitleMatch = a.title.toLowerCase().includes(lowerQuery);
      const bTitleMatch = b.title.toLowerCase().includes(lowerQuery);
      if (aTitleMatch && !bTitleMatch) return -1;
      if (!aTitleMatch && bTitleMatch) return 1;
      return 0;
    });

    setResults(allResults);
  }, [query]);

  const handleKeyDown = (e: React.KeyboardEvent) => {
    if (e.key === "Escape") {
      onClose();
    } else if (e.key === "Enter" && results.length > 0) {
      router.push(results[0].href);
      onClose();
    }
  };

  const handleResultClick = (href: string) => {
    router.push(href);
    onClose();
  };

  return (
    <>
      <div
        className="fixed inset-0 bg-black/50 z-50"
        onClick={onClose}
        aria-hidden="true"
      />
      <div className="fixed inset-x-4 top-20 mx-auto max-w-2xl bg-white rounded-lg shadow-xl z-50 max-h-[80vh] flex flex-col">
        <div className="flex items-center gap-3 p-4 border-b border-gray-200">
          <Search className="w-5 h-5 text-gray-400" aria-hidden="true" />
          <input
            ref={inputRef}
            type="text"
            value={query}
            onChange={(e) => setQuery(e.target.value)}
            onKeyDown={handleKeyDown}
            placeholder="Tìm kiếm kiến thức hoặc công cụ..."
            className="flex-1 outline-none text-base"
            aria-label="Tìm kiếm"
          />
          <button
            type="button"
            onClick={onClose}
            className="p-1 rounded hover:bg-gray-100 focus-ring"
            aria-label="Đóng"
          >
            <X className="w-5 h-5 text-gray-400" aria-hidden="true" />
          </button>
        </div>

        <div className="overflow-y-auto flex-1">
          {query.trim() && results.length === 0 && (
            <div className="p-8 text-center text-gray-500">
              <p>Không tìm thấy kết quả cho &quot;{query}&quot;</p>
              <p className="text-sm mt-2">
                Thử tìm kiếm với từ khóa khác
              </p>
            </div>
          )}

          {!query.trim() && (
            <div className="p-8 text-center text-gray-500">
              <p>Nhập từ khóa để tìm kiếm...</p>
            </div>
          )}

          {results.length > 0 && (
            <div className="p-2">
              <div className="text-xs font-semibold text-gray-500 uppercase tracking-wide px-3 py-2">
                {results.length} kết quả
              </div>
              <ul className="space-y-1">
                {results.map((result) => (
                  <li key={result.href}>
                    <button
                      type="button"
                      onClick={() => handleResultClick(result.href)}
                      className="w-full text-left px-3 py-3 rounded-lg hover:bg-gray-100 focus-ring flex items-start gap-3"
                    >
                      {result.category === "knowledge" ? (
                        <BookOpen
                          className="w-5 h-5 text-blue-600 mt-0.5 flex-shrink-0"
                          aria-hidden="true"
                        />
                      ) : (
                        <Calculator
                          className="w-5 h-5 text-green-600 mt-0.5 flex-shrink-0"
                          aria-hidden="true"
                        />
                      )}
                      <div className="flex-1 min-w-0">
                        <div className="font-medium text-gray-900">
                          {highlightText(result.title, query)}
                        </div>
                        {result.description && (
                          <div className="text-sm text-gray-600 mt-1">
                            {highlightText(result.description, query)}
                          </div>
                        )}
                        {result.tags && result.tags.length > 0 && (
                          <div className="flex flex-wrap gap-1 mt-1">
                            {result.tags.slice(0, 3).map((tag) => (
                              <span
                                key={tag}
                                className="text-xs px-1.5 py-0.5 bg-gray-100 text-gray-600 rounded"
                              >
                                {tag}
                              </span>
                            ))}
                          </div>
                        )}
                        <div className="text-xs text-gray-400 mt-1">
                          {result.href}
                        </div>
                      </div>
                    </button>
                  </li>
                ))}
              </ul>
            </div>
          )}
        </div>

        <div className="p-3 border-t border-gray-200 text-xs text-gray-500">
          <kbd className="px-2 py-1 bg-gray-100 rounded">Esc</kbd> để đóng •{" "}
          <kbd className="px-2 py-1 bg-gray-100 rounded">Enter</kbd> để chọn
        </div>
      </div>
    </>
  );
}
